### PR TITLE
Add support for additional vector/haproxy service spec options

### PR DESCRIPTION
**Pull request by r0b3r4**
- externalTrafficPolicy
- loadBalancerIP
- ipFamilyPolicy
- ipFamilies

### CODE REVIEW

File: charts/vector/values.yaml [hunk 1]
```diff
@@ -223,6 +223,14 @@ service:
...
   # service.ports -- Manually set the Service ports, overriding automated generation of Service ports.
   ports: []
+  # service.externalTrafficPolicy -- Specify the [externalTrafficPolicy](https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#preserving-the-client-source-ip)
+  externalTrafficPolicy: ""
+  # service.loadBalancerIP -- Specify the [loadBalancerIP](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer)
+  loadBalancerIP: ""
+  # service.ipFamilyPolicy -- Configure [IPv4/IPv6 dual-stack](https://kubernetes.io/docs/concepts/services-networking/dual-stack/)
+  ipFamilyPolicy: ""
+  # service.ipFamilies -- Configure [IPv4/IPv6 dual-stack](https://kubernetes.io/docs/concepts/services-networking/dual-stack/)
+  ipFamilies: []
```
Comment (spencergilbert) [inline charts/vector/values.yaml:233]:
```suggestion
  # service.externalTrafficPolicy -- Specify the [externalTrafficPolicy](https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#preserving-the-client-source-ip).
  externalTrafficPolicy: ""
  # service.loadBalancerIP -- Specify the [loadBalancerIP](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer).
  loadBalancerIP: ""
  # service.ipFamilyPolicy -- Configure [IPv4/IPv6 dual-stack](https://kubernetes.io/docs/concepts/services-networking/dual-stack/).
  ipFamilyPolicy: ""
  # service.ipFamilies -- Configure [IPv4/IPv6 dual-stack](https://kubernetes.io/docs/concepts/services-networking/dual-stack/).
  ipFamilies: []
```

File: charts/vector/values.yaml [hunk 2]
```diff
@@ -463,6 +471,14 @@ haproxy:
...
     # haproxy.service.ports -- Manually set HAPRoxy's Service ports, overrides automated generation of Service ports.
     ports: []
+    # haproxy.service.externalTrafficPolicy -- Specify the [externalTrafficPolicy](https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#preserving-the-client-source-ip)
+    externalTrafficPolicy: ""
+    # haproxy.service.loadBalancerIP -- Specify the [loadBalancerIP](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer)
+    loadBalancerIP: ""
+    # haproxy.service.ipFamilyPolicy -- Configure [IPv4/IPv6 dual-stack](https://kubernetes.io/docs/concepts/services-networking/dual-stack/)
+    ipFamilyPolicy: ""
+    # haproxy.service.ipFamilies -- Configure [IPv4/IPv6 dual-stack](https://kubernetes.io/docs/concepts/services-networking/dual-stack/)
+    ipFamilies: []
```
Comment (spencergilbert) [inline charts/vector/values.yaml:481]:
```suggestion
    # haproxy.service.externalTrafficPolicy -- Specify the [externalTrafficPolicy](https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#preserving-the-client-source-ip).
    externalTrafficPolicy: ""
    # haproxy.service.loadBalancerIP -- Specify the [loadBalancerIP](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer).
    loadBalancerIP: ""
    # haproxy.service.ipFamilyPolicy -- Configure [IPv4/IPv6 dual-stack](https://kubernetes.io/docs/concepts/services-networking/dual-stack/).
    ipFamilyPolicy: ""
    # haproxy.service.ipFamilies -- Configure [IPv4/IPv6 dual-stack](https://kubernetes.io/docs/concepts/services-networking/dual-stack/).
    ipFamilies: []
```

**Review by spencergilbert — CHANGES_REQUESTED** [review]
Thanks again for the contribution, two small nits I added suggestions for - and it looks like `helm-docs` needs rerunning before it'll pass CI.

File: charts/vector/values.yaml (hunk 1 above)
Comment (r0b3r4) [reply charts/vector/values.yaml:233]: Are you sure these defaults will does not break backward compability?

**Comment by spencergilbert** [comment]
@r0b3r4 could you check what `helm-docs` version you used locally? It seems like the README you generated wasn't matching CI. I merged in `develop` and regenerated the docs, and it seems happy now.

```bash
helm-docs version 1.11.0
```

**Review by spencergilbert — APPROVED** [review]

**Comment by r0b3r4** [comment]
Hey ya!

Can you please tell when the next release in which these changes could enter?

**Comment by spencergilbert** [comment]
Sorry for the delay, I expect we'll cut the release this week though.

**Pull request merged**